lib/flags.ml

```diff
@@ -11,17 +11,14 @@
 (* If [restore] is false, whenever [f] modifies the ref, we will
    preserve the modification. *)
 let with_modified_ref ?(restore=true) r nf f x =
```

Comment:
with_options a few lines below looks like similar code

Reply:
Yes, I ran out of cycles to make this safe; will try if I have time, but indeed I was never able to trigger a coherence problem with this code; feel free to push and fix this TODO.